File: week01/1085.py
# 구해야 할 것 : 직사각형의 경계선까지의 최소거리
# 최소 거리 -> 가로변 까지의 거리(h - y), 세로변까지의 거리(w - x), x축까지의 거리, y축 까지의 거리 중 더 작은 값

import sys
input = sys.stdin.readline

x, y, w, h = map(int, input().split())
print(min(h - y , w - x, x, y))

File: week01_JS/조건문/1085.js
// 주어지는 (x,y) 에서 각각 x축, y축, 사각형 세로와의 거리, 사각형 가로와의 거리 중 최소를 리턴
const fs = require("fs")
const input = fs.readFileSync('input.txt').toString().trim().split(' ')

input_list = input.map((elems) => parseInt(elems))
let [x, y, w, h] = input_list
result_list = [x, y, w - x, h - y]


console.log(Math.min(...result_list))
